osu.Game/Skinning/LegacySkin.cs

```diff
@@ -376,6 +376,7 @@ public override Drawable GetDrawableComponent(ISkinComponent component)
                                     hitError.Anchor = Anchor.BottomCentre;
                                     hitError.Origin = Anchor.CentreLeft;
                                     hitError.Rotation = -90;
+                                    if (hitError is BarHitErrorMeter barHitError) barHitError.OrientIcons();
```

Comment:
I disagree with exposing the `OrientIcons()` method externally and calling it in this manner. It breaks encapsulation; the icons being kept upright is a part of the logic of the bar meter _itself_, and so the meter hold the entirety of the logic that does so and not rely on some external call.

A solution that fixes this (and also another issue that this PR doesn't solve, which is that the icons will not remain upright if a skin is edited with the skin editor) would be to put the contents of `OrientIcons()` in the bar meter's `Update()`. Doing that sort of thing is usually a last-resort solution (because it runs every frame) but I think it may be defensible in this context as setting `Rotation` is a no-op if the value doesn't change.

Reply:
Rotating in `Update` is fine in this case, and also what I would do

Reply:
Ideally it would run whenever the `BarHitErrorMeter.Rotation` is set, but I didn't see an obvious way of subscribing to that

Reply:
That's because there isn't one. Fine to run every frame either way as stated.

Reply:
Is hooking into `OnNewJudgement()` an alright compromise, to keep rotation updated and not to call it every frame?

Reply:
It is not. See above for reasoning (skin editor).

Reply:
Ok, change pushed